src/components/transactions/Switch/BaseSwitchModalContent.tsx

```diff
@@ -413,12 +489,91 @@ export const BaseSwitchModalContent = ({
     }
   }, [txError]);
 
+  // Compute HF effect of withdrawing inputAmount (copied from SwitchModalTxDetails)
+  const { hfEffectOfFromAmount, hfAfterSwap } = useMemo(() => {
+    try {
+      if (!poolReserve || !userReserve || !extendedUser || !switchRates || !targetReserve)
+        return { hfEffectOfFromAmount: '0' };
+
+      // Amounts in human units (mirror SwitchModalTxDetails: intent uses destSpot, market uses destAmount)
+      const fromAmount = normalizeBN(switchRates.srcAmount, switchRates.srcDecimals).toString();
+      const toAmountRaw = normalizeBN(
+        switchRates.provider === 'cowprotocol' ? switchRates.destSpot : switchRates.destAmount,
+        switchRates.destDecimals
+      ).toString();
+      const toAmountAfterSlippage = valueToBigNumber(toAmountRaw)
+        .multipliedBy(1 - safeSlippage)
+        .toString();
+
+      const { hfEffectOfFromAmount, hfAfterSwap } = calculateHFAfterSwap({
+        fromAmount,
+        fromAssetData: poolReserve,
+        fromAssetUserData: userReserve,
+        user: extendedUser,
+        toAmountAfterSlippage: toAmountAfterSlippage,
+        toAssetData: targetReserve,
+      });
+
+      return {
+        hfEffectOfFromAmount: hfEffectOfFromAmount.toString(),
+        hfAfterSwap: hfAfterSwap.toString(),
+      };
+    } catch {
+      return { hfEffectOfFromAmount: '0', hfAfterSwap: undefined };
+    }
+  }, [
+    poolReserve,
+    userReserve,
+    extendedUser,
+    inputAmount,
+    targetReserve,
+    switchRates,
+    safeSlippage,
+  ]);
+
+  const isHFLow = useMemo(() => {
+    if (!hfAfterSwap) return false;
+
+    const hfNumber = new BigNumber(hfAfterSwap);
+
+    if (hfNumber.lt(0)) return false;
+
+    return hfNumber.lt(1.05);
```

Comment:
Can we label this variable

Reply:
const HEALTH_FACTOR_DANGER_THRESHOLD = 1.05; // 5% buffer above liquidation (1.0)

Reply:
maybe less extreme const LIQUIDATION_SAFETY_THRESHOLD = 1.05;